stock_custom_exchange_rate_date/models/stock_move.py

```diff
@@ -0,0 +1,27 @@
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _get_price_unit(self):
+        """Take into account custom rate date, if provided"""
+        line = self.purchase_line_id
+        if (
+            not self.picking_id.exchange_rate_date
+            or not line
```

Comment:
There is a new condition on the original method:

`self.origin_returned_move_id`

This was not on the previous version, so I think it should be included here as condition to return the super. If this is a return, the price should be taken from the returned move.

Reply:
Done